src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java

```diff
@@ -0,0 +1,343 @@
+  }
+
+  /**
+   * Lists all the files in the {@code directoryPath} which match the given {@code matcher}.
+   * <p>
+   * If the listing encounters a directory, the output list will include its contents depending on the value of the
+   * {@code recursive} parameter. If {@code recursive} is enabled, then all the files in that directory will be listed immediately
+   * after their parent directory.
+   * <p>
+   *
+   * @param config        the config that is parameterizing this operation
+   * @param directoryPath the path to the directory to be listed
+   * @param recursive     whether to include the contents of sub-directories. Defaults to false.
+   * @param matchWith     a matcher used to filter the output list
+   * @param subsetList    parameter group that lets you obtain a subset of the results
+   * @return a {@link PagingProvider} of {@link Result} objects each one containing each file's content in the payload and metadata in the
+   * attributes
+   * @throws IllegalArgumentException if {@code directoryPath} points to a file which doesn't exist or is not a directory
+   */
+  protected PagingProvider<FileSystem, Result<String, FtpFileAttributes>> doPagedList(FileConnectorConfig config,
+                                                                                      String directoryPath,
+                                                                                      boolean recursive,
+                                                                                      FileMatcher matchWith,
+                                                                                      StreamingHelper streamingHelper,
+                                                                                      SubsetList subsetList) {
+    return new PagingProvider<FileSystem, Result<String, FtpFileAttributes>>() {
+
+      private List<Result<String, FtpFileAttributes>> files;
+      private Iterator<Result<String, FtpFileAttributes>> filesIterator;
+      private final AtomicBoolean initialised = new AtomicBoolean(false);
+
+      @Override
+      public List<Result<String, FtpFileAttributes>> getPage(FileSystem connection) {
+        if (initialised.compareAndSet(false, true)) {
+          initializePagingProvider(connection);
+        }
+        List<Result<String, FtpFileAttributes>> page = new LinkedList<>();
+        for (int i = 0; i < LIST_PAGE_SIZE && filesIterator.hasNext(); i++) {
+          Result<String, FtpFileAttributes> result = filesIterator.next();
+          page.add((Result.<String, FtpFileAttributes>builder().attributes(result.getAttributes().get())
+              .output(result.getOutput())
+              .mediaType(result.getMediaType().orElse(null))
+              .attributesMediaType(result.getAttributesMediaType().orElse(null))
+              .build()));
+        }
+        return page;
+      }
+
+      private void initializePagingProvider(FileSystem connection) {
+        connection.changeToBaseDir();
```

Comment:
What if the directoryPath is an absolute path? Why do this?

Reply:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.